core-blocks/image/block.js

```diff
@@ -389,14 +386,12 @@ class ImageBlock extends Component {
 	}
 }
 
-export default compose( [
-	withEditorSettings(),
-	withSelect( ( select, props ) => {
-		const { getMedia } = select( 'core' );
-		const { id } = props.attributes;
+export default withSelect( ( select, props ) => {
+	const { getMedia } = select( 'core' );
+	const { id } = props.attributes;
 
-		return {
-			image: id ? getMedia( id ) : null,
-		};
-	} ),
-] )( ImageBlock );
+	return {
+		settings: select( 'core/editor' ).getEditorSettings(),
```

Comment:
Minor: This change assume settings can now be dynamic, so we will re-render each time settings change, we just use maxWidth from settings so maybe we can just pass maxWidth now instead of all the settings.

Reply:
Good point, though there's not real change compared to the previous version in term of rerendering. The previous version, the settings were also dynamic since it was using the new context API.